### DISCUSSION TITLE
Can I change the api route folder

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [vercel/next.js#22256] [Help]
Our nginx is configured to any request like `/api/*` will point to the backend server.

So the issue is I and not able to use api routes with next js because if any we make any endpoint like `/api/something` it will not allow to handle to a next js node server.

Anyone knows how to solve it.

**Comment by mkrtchian** [comment]
You can use [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) and have your API routes available from another path:

In `next.config.js`:
```javascript
module.exports = {
  async rewrites() {
    return [{ source: "/custom-api/:path*", destination: "/api/:path*" }];
  },
};
```

**Reply by rajatdhoot123** [reply]
Not working it was detecting dynamic pages under `/page` folder

![image](https://user-images.githubusercontent.com/18528826/108205993-cc5c0280-714b-11eb-97d0-3ab357a4252c.png)

Path trying to access:
`/custom-api/hello`

here is my folder structure

![image](https://user-images.githubusercontent.com/18528826/108206087-ec8bc180-714b-11eb-90cc-34671909c722.png)

**Reply by michrome** [reply]
`api/hello.js` should be `pages/api/hello.js` for the rewrite to work, not the root of your project.

**Reply by rajatdhoot123** [reply]
Thanks, @michrome it's working

**Reply by juanektbb** [reply]
This makes `req` and `res` to be empty objects {}, so you can't use them in your API. 
Anyone knows how to resolve it?
